### PR TITLE
simplify run and concurrent usage calculations

### PR DESCRIPTION
**Pull request by infinitewarp**
for https://github.com/cloudigrade/cloudigrade/issues/948

These commits should result in the following changes:

- run objects are no longer automatically rebuilt when we generate new events
- concurrentusage tasks are no longer automatically created when we generate new runs
- run calculation logic is simplified (likely fixing some preexisting bugs)
- run objects are created as needed on a periodic task for each user
- concurrentusage objects are created as needed on a periodic task for each user

### CODE REVIEW

File: cloudigrade/api/tests/util/test_get_max_concurrent_usage.py [hunk 1]
```diff
@@ -69,10 +44,10 @@ def test_single_rhel_run_result(self, mock_schedule_concurrent_calc):
     @util_helper.clouditardis(util_helper.utc_dt(2019, 5, 3, 0, 0, 0))
     def test_single_rhel_run_no_result(self):
         """
-        Test with a single RHEL instance requesting results in futures.
+        Test with a single RHEL instance requesting results in the future.
 
-        Also assert that no ConcurrentUsage is saved to the database because the
-        requested date is in the future, and we cannot know that data yet.
+        We expect None because the we refuse to calculate data for future dates.
```
Comment (katherine-black) [inline cloudigrade/api/tests/util/test_get_max_concurrent_usage.py:49]: The we?

**Review by katherine-black — APPROVED** [review]

File: cloudigrade/api/tests/util/test_get_max_concurrent_usage.py (hunk 1 above)
Comment (infinitewarp) [reply cloudigrade/api/tests/util/test_get_max_concurrent_usage.py]: Whoops! Good catch. :) I just amended the specific commit that introduced that change and force-pushed everything back up.

**Review by katherine-black — APPROVED** [review]

**Review by abellotti — APPROVED** [review]
took a while to go through, but yeah, LGTM !!

**Comment by infinitewarp** [comment]
demo recording of the updated run and concurrent usage calculation process: https://asciinema.org/a/427887

**Pull request merged**